addons/l10n_it_edi_sdi_coop/models/account_edi_format.py

```diff
@@ -0,0 +1,268 @@
+# -*- coding:utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, _
+
+from lxml import etree
+import base64
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.asymmetric import padding
+from cryptography.fernet import Fernet
+
+
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountEdiFormat(models.Model):
+    _inherit = 'account.edi.format'
+
+    # -------------------------------------------------------------------------
+    # Helpers
+    # -------------------------------------------------------------------------
+
+    def _l10n_it_edi_get_proxy_user(self, company):
+        return self.env['account_edi_proxy_client.user'].search([
+            ('company_id', '=', company.id),
+            ('edi_identification', '=', company.l10n_it_codice_fiscale)], limit=1)
+
+    def _l10n_it_edi_decrypt_data(self, data, key, proxy_user):
+        private_key = serialization.load_pem_private_key(
```

Comment:
Too much stuff on one line, not enough stuff on one line :( If I split parameters, I usually prefer to have one parameter per line as it's easier to read rather than having an arbitrary number of parameter per line.